platform/posix/transport/src/openssl_posix.c

```diff
@@ -778,8 +782,20 @@ int32_t Openssl_Send( NetworkContext_t * pNetworkContext,
         {
             sslError = SSL_get_error( pOpensslParams->pSsl, bytesSent );
 
-            LogError( ( "Failed to send data over network: SSL_write of OpenSSL failed: "
-                        "ErrorStatus=%s.", ERR_reason_error_string( sslError ) ) );
+            if( ( sslError == SSL_ERROR_WANT_WRITE ) || ( sslError == SSL_ERROR_WANT_READ ) )
```

Comment:
If I recall this wasn't checked here because `SSL_write` was blocking and these values can't be returned. Is this necessary, @yourslab ?

Reply:
Yah, this is blocking, so we don't expect to get these errors. However, we actually have renegotiation activated because of the `SSL_MODE_AUTO_RETRY` flag. In that case, we only want to check `SSL_ERROR_WANT_READ`.

Reply:
Based on the documentation of [OpenSSL_write](https://www.openssl.org/docs/man1.0.2/man3/SSL_write.html), `SSL_ERROR_WANT_READ` can occur in a blocking case (unless the the auto retry mode is set)

Reply:
If `SSL_MODE_AUTO_RETRY` is set, then we need to check `SSL_ERROR_WANT_READ`. `SSL_ERROR_WANT_WRITE` check should be removed though.

Reply:
Oops, didn't notice that the auto retry mode was set. Have removed the check for `SSL_ERROR_WANT_READ`.

Reply:
However, the auto retry requires us to check `SSL_ERROR_WANT_READ`. What needs to be removed is `SSL_ERROR_WANT_WRITE`.

Reply:
Okay, updated.